src/libs/actions/User.ts

```diff
@@ -888,6 +893,124 @@ function playSoundForMessageType(pushJSON: OnyxServerUpdate[]) {
     });
 }
 
+// Holds a map of all the PINGs that have been sent to the server and when they were sent
+// Once a PONG is received, the event data will be removed from this map.
+type PingPongTimestampMap = Record<string, number>;
+let pingIDsAndTimestamps: PingPongTimestampMap = {};
+
+function subscribeToPusherPong() {
+    // If there is no user accountID yet (because the app isn't fully setup yet), the channel can't be subscribed to so return early
+    if (currentUserAccountID === -1) {
+        return;
+    }
+
+    PusherUtils.subscribeToPrivateUserChannelEvent(Pusher.TYPE.PONG, currentUserAccountID.toString(), (pushJSON) => {
+        Log.info(`[Pusher PINGPONG] Received a PONG event from the server`, false, pushJSON);
+        const pongEvent = pushJSON as Pusher.PingPongEvent;
+        // First, check to see if the PONG event is in the pingIDsAndTimestamps map
+        // It's OK if it doesn't exist. The client was maybe refreshed while still waiting for a PONG event, in which case it might
+        // receive the PONG event but has already lost it's memory of the PING.
+        const pingEventTimestamp = pingIDsAndTimestamps[pongEvent.pingID];
+        if (!pingEventTimestamp) {
+            return;
+        }
+
+        // Calculate the latency between the client and the server
+        const latency = Date.now() - Number(pingEventTimestamp);
+        Log.info(`[Pusher PINGPONG] The event took ${latency} ms`);
+
+        // Remove the event from the map
+        delete pingIDsAndTimestamps[pongEvent.pingID];
+        Performance.markEnd(CONST.TIMING.PUSHER_PING_PONG);
```

Comment:
@adamgrzybowski I wanted to CC you on this change to ensure I'm not missing anything with the performance tracking library. Happy to give you more context if you need it. I mostly just want to ensure I'm using this correctly and I'd like to build a Grafana chart for this.

cc @mountiny

Reply:
Also missed it the first time I read the comment 🤦 pinged Adam in Slack now

Reply:
Hey @tgolen, here's a breakdown:
- `Performance` module is mostly used for local measurements
- `Timing` module is reporting to external systems

If you want to send to Grafana, you can use the Timing module as per [this example](https://github.com/Expensify/App/blob/bdf216ee216f874bef2ed235ba14407f865953f2/src/components/LHNOptionsList/OptionRowLHN.tsx#L172). `Timing.start()/end()` will ensure it's being sent.

I have plans for proposing a vision for these 2 going forward in a more unified way, but ^this is how they work now :)

Reply:
Aha, perfect. I'll swap this out to use the timing module. Thanks!

Reply:
OK, I updated this and see it in the logs now:

```
Timing:development.new.expensify.pusher_ping_pong 949.5
```